kubectl-plugin/pkg/util/client/client_test.go

```diff
@@ -6,12 +6,101 @@ import (
 
 	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util"
 	"github.com/stretchr/testify/assert"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
 	dynamicFake "k8s.io/client-go/dynamic/fake"
 	kubeFake "k8s.io/client-go/kubernetes/fake"
 )
 
+func TestGetKubeRayOperatorVersion(t *testing.T) {
+	kubeObjects := []runtime.Object{
+		&appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "kuberay-operator-helm-chart",
+				Namespace: "default",
+				Labels: map[string]string{
+					"app.kubernetes.io/name": "kuberay-operator",
+				},
+			},
+			Spec: appsv1.DeploymentSpec{
+				Template: corev1.PodTemplateSpec{
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Image: "kuberay/operator:v0.5.0",
+							},
+						},
+					},
+				},
+			},
+		},
+		&appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "kuberay-operator-kustomize",
+				Namespace: "test",
+				Labels: map[string]string{
+					"app.kubernetes.io/name": "kuberay",
+				},
+			},
+			Spec: appsv1.DeploymentSpec{
+				Template: corev1.PodTemplateSpec{
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Image: "kuberay/operator:v0.6.0",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	tests := []struct {
+		name            string
+		expectedVersion string
+		expectedError   string
+		kubeObjects     []runtime.Object
+	}{
+		{
+			name:            "kubeRay operator not found",
+			expectedVersion: "",
+			expectedError:   "no KubeRay operator deployments found in any namespace",
+			kubeObjects:     nil,
+		},
+		{
+			name:            "find kubeRay operator version for helm chart",
+			expectedVersion: "v0.5.0",
+			expectedError:   "",
+			kubeObjects:     kubeObjects[:1],
```

Comment:
I find it is a little bit hard to read this. How about separate them to 2 variables like `helmKubeObjects` and `kustomizeKubeObjects`?

Reply:
Sounds good.